swap/src/protocol/bob/swap.rs

```diff
@@ -134,8 +139,9 @@ async fn next_state(
                 },
             );
 
-            let (..) = bitcoin_wallet.broadcast(signed_tx, "lock").await?;
-
+            // Wait until tx_lock was confirmed
+            subscription.wait_until_final().await?;
```

Comment:
We cannot wait for a confirmation before transitioning into the next state.

If we stop the app after we have published the Bitcoin transaction but before the transaction is confirmed, we'll
1. attempt to publish an already existing transaction into the mempool after resuming which might be rejected by the Bitcoin node
2. assume we haven't locked our funds even though our transaction has been published which means we can get punished without noticing